kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -111,11 +117,19 @@ class BatchJobSubmission(session: KyuubiBatchSessionImpl, batchRequest: BatchReq
   }
 
   private def submitBatchJob(): Unit = {
+    var appStatusFirstUpdated = false
     try {
       info(s"Submitting $batchType batch job: $builder")
       val process = builder.start
-      var applicationStatus = currentApplicationState
+      applicationStatus = currentApplicationState
       while (!applicationFailed(applicationStatus) && process.isAlive) {
+        if (!appStatusFirstUpdated && applicationStatus.isDefined) {
+          session.sessionManager.updateBatchStateAppInfo(
+            batchId,
+            getStatus.state,
+            applicationStatus.getOrElse(Map.empty))
```

Comment:
assert(applicationStatus.isDefined)
applicationStatus.get

Reply:
we have check `applicationStatus.isDefined` in if condition, let me use `get` directly.